Add docs for new fan trigger: on_speed_set (#1447)
OttoWinter/esphomedocs: components/fan/index.rst

  when the fan is turned on. See :ref:`fan-on_turn_on_off_trigger`.
- **on_turn_off** (*Optional*, :ref:`Action <config-action>`): An automation to perform
  when the fan is turned off. See :ref:`fan-on_turn_on_off_trigger`.
- **on_speed_set** (*Optional*, :ref:`Action <config-action>`): An automation to perform

>   when the fan speed is set/changed. See :ref:`fan-on_speed_set_trigger`.

Reviewer: ```suggestion
  when the fan speed is set/changed. See :ref:`fan-on_speed_set_trigger`.

```

            id: fan_1
        # Shorthand:

>         - fan.turn_on: fan_1

Reviewer: ```suggestion
        - fan.turn_on: fan_1

```

        - logger.log: "Fan Turned On!"
        on_turn_off:
        - logger.log: "Fan Turned Off!"

> .. _fan-on_speed_set_trigger:

Reviewer: ```suggestion

.. _fan-on_speed_set_trigger:
```

        on_turn_off:
        - logger.log: "Fan Turned Off!"

.. _fan-on_speed_set_trigger:

``fan.on_speed_set`` Trigger
****************************************************

This trigger is activated each time the fan speed is changed. Itwill fire when the speed is either set via API e.g. in Home Assstant or locally by an automation or a lambda function. 

.. code-block:: yaml

    fan:
      - platform: speed # or any other platform
        # ...
        on_speed_set:

>         - logger.log: "Fan Speed was changed!"

Reviewer: ```suggestion
        - logger.log: "Fan Speed was changed!"

```

        on_turn_off:
        - logger.log: "Fan Turned Off!"

.. _fan-on_speed_set_trigger:

``fan.on_speed_set`` Trigger
****************************************************

> This trigger is activated each time the fan speed is changed. Itwill fire when the speed is either set via API e.g. in Home Assstant or locally by an automation or a lambda function. 

Reviewer: ```suggestion
This trigger is activated each time the fan speed is changed. It will fire when the speed is either set via API e.g. in Home Assistant or locally by an automation or a lambda function. 
```

.. code-block:: yaml

    fan:
      - platform: speed # or any other platform
        # ...
        on_speed_set:
        - logger.log: "Fan Speed was changed!"

>         

Reviewer: ```suggestion

```
